### PR TITLE
CBG-3924: Add mocked config and API for audit logging

### PR DESCRIPTION
**Pull request by bbrks**
CBG-3924

Unblocks Capella work since they can set the new config and reference new API endpoints.
- Exposes `SGAuditEvents` for visibility from handler.
- Adds mocked audit logging API endpoints
- Adds unwired bootstrap config options for audit log output
- Adds unwired dbconfig options for per-db audit log settings
- Extends flag config support for additional flag types and pointer/non-pointers

## Toy build
https://mobile.jenkins.couchbase.com/view/Sync_Gateway/job/sgw-toy-build/258/ 

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [ ] `GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/000/

### CODE REVIEW

**Review by torcolvin — APPROVED** [review]
LGTM, is this worth adding to openapi docs?

**Comment by bbrks** [comment]
@JRascagneres should I wait until you've tried this toy build before I merge?

**Comment by JRascagneres** [comment]
@bbrks Up to you, might be a 'bit' away yet before I can test out all the endpoints.

**Comment by bbrks** [comment]
Can you at least review then to make sure the config and endpoints are as you'd expect?

File: docs/api/components/schemas.yaml
Comment (torcolvin) [inline docs/api/components/schemas.yaml:1847]: Should this say that certain events will be enabled by default and have to be explicitly disabled?

File: docs/api/paths/admin/db-_config-audit.yaml
Comment (torcolvin) [inline docs/api/paths/admin/db-_config-audit.yaml:15]: Do we have a ticket to decide RBAC roles?

**Review by torcolvin — APPROVED** [review]

File: docs/api/components/schemas.yaml
Comment (bbrks) [reply docs/api/components/schemas.yaml:1847]: Had discussions with PM earlier this week. The plan is to have this represent the full set of enabled events. It contains the set of default events, and you can disable by removing from this list.

File: docs/api/paths/admin/db-_config-audit.yaml
Comment (bbrks) [reply docs/api/paths/admin/db-_config-audit.yaml:15]: Following up in PRD

**Review by JRascagneres — APPROVED** [review]

**Pull request merged**
